tests/e2e/test_fee_receiver_flow.py

```diff
@@ -0,0 +1,102 @@
+import boa
+import pytest
+from tests.utils.constants import ZERO_ADDRESS
+
+
+@pytest.fixture(scope="module")
+def market_type():
+    return "lending"
+
+
+def test_fee_receiver_flow(
+    controller, factory, admin, collateral_token, borrowed_token, amm
+):
+    """
+    Test flow for setting and updating custom fee receivers for Llamalend markets.
+
+    Context:
+    This feature allows Llamalend markets to request the DAO for a custom fee receiver.
+    This feature has been designed for revenue sharing designs where the custom fee receiver
+    would be a helper contract responsible of splitting admin fees between multiple parties.
+
+    Implementation notes:
+    - By default all markets have the same fee_receiver (most likely DAO)
+    - Markets still fetch the fee receiver from the factory
+    - Factory acts as the registry of fee receivers
+    - Factory now responds to the fee_receiver getter conditionally depending on the msg.sender
+      (since controllers don't specify the first argument, they get the one corresponding to their mapping).
+    """
+    # Setup
+    collateral_amount = 10**18
+    debt_amount = 10**18
+    bands = 10
+
+    boa.deal(collateral_token, boa.env.eoa, collateral_amount * 10)
+    boa.deal(borrowed_token, boa.env.eoa, debt_amount * 10)
+
+    collateral_token.approve(controller, collateral_amount * 10)
+
+    # Create loan
+    controller.create_loan(collateral_amount, debt_amount, bands)
+
+    # Time travel to accrue interest
+    boa.env.time_travel(seconds=365 * 86400)
+
+    # Trigger debt update to ensure fees are pending (if admin_percentage > 0)
+    controller.add_collateral(0, boa.env.eoa)
```

Comment:
0 amount doesn't trigger

```
@external
def add_collateral(_collateral: uint256, _for: address = msg.sender):
    """
    @notice Add extra collateral to avoid bad liqidations
    @param _collateral Amount of collateral to add
    @param _for Address to add collateral for
    """
    if _collateral == 0:
        return
    self._add_collateral_borrow(_collateral, 0, _for, False, _for != msg.sender)
    tkn.transfer_from(COLLATERAL_TOKEN, msg.sender, AMM.address, _collateral)
    self._save_rate()
```